src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/AgencyLocation.java

```diff
@@ -41,6 +46,18 @@
 @Table(name = "AGENCY_LOCATIONS")
 @ToString(of = {"id", "description"})
 @With
+@NamedEntityGraph(
```

Comment:
This will result in a Cartesian Product but will still perform better than the n * 3 + 1 queries since we know real world data is constrained to a single address and 2 phone numbers so it will read 127 * 1 * 2 * 2 which is diddly squat